### PR TITLE
Support localspecrepo if pass in this parameter

### PR DESCRIPTION
**Pull request by raych1**
- Add `LocalSpecRepoPath` to sync script
- Code style fix

@weshaggard , this PR shouldn't be merged until all sdk language repos adopt new common scripts.

Fixing this issue in sync script.
https://github.com/Azure/azure-sdk-tools/issues/6279

### CODE REVIEW

File: eng/common/scripts/TypeSpec-Project-Sync.ps1 [hunk 1]
```diff
@@ -13,115 +15,133 @@ Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
...
-                    throw "Unknown git remote format found: $remote"
-                }
-            }
+  Push-Location $ProjectDirectory
+  $result = ""
+  try {
+    $gitRemotes = (git remote -v)
+    foreach ($remote in $gitRemotes) {
+      Write-Host "Checking remote $remote"
+      if ($remote.StartsWith("origin") -or $remote.StartsWith("main")) {
```
Comment (weshaggard) [inline eng/common/scripts/TypeSpec-Project-Sync.ps1:50]: What context do we need to support "main" as a remote?

File: eng/common/scripts/TypeSpec-Project-Sync.ps1 [hunk 2]
```diff
@@ -13,115 +15,133 @@ Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
...
-        }
-        UpdateSparseCheckoutFile $specSubDirectory $configuration["additionalDirectories"]
-        git checkout $configuration["commit"]
-        if ($LASTEXITCODE) { exit $LASTEXITCODE }
-    }
-    finally {
-        Pop-Location
+# use local spec repo if provided
+if ($LocalSpecRepoPath) {
+  $specCloneDir = $LocalSpecRepoPath
```
Comment (weshaggard) [inline eng/common/scripts/TypeSpec-Project-Sync.ps1:106]: Do we also need to checkout the correct branch or at least validate that it matches?

**Comment by weshaggard** [comment]
@raych1 can you please link the issue you are fixing with this in the description?

File: eng/common/scripts/TypeSpec-Project-Sync.ps1 (hunk 1 above)
Comment (raych1) [reply eng/common/scripts/TypeSpec-Project-Sync.ps1:50]: These are the remotes in the pipeline run of test repo which doesn't have `origin`. That's why I added main.
```
10:29:05.890 cmdout 	[generate.py] Checking remote integration	https://x-access-token:***@github.com/openapi-env-test/azure-sdk-for-java (fetch)
10:29:05.891 cmdout 	[generate.py] Checking remote integration	https://x-access-token:***@github.com/openapi-env-test/azure-sdk-for-java (push)
10:29:05.891 cmdout 	[generate.py] Checking remote main	https://x-access-token:***@github.com/openapi-env-test/azure-sdk-for-java (fetch)
10:29:05.891 cmdout 	[generate.py] Checking remote main	https://x-access-token:***@github.com/openapi-env-test/azure-sdk-for-java (push)
10:29:05.891 cmdout 	[generate.py] Checking remote secondary	https://x-access-token:***@github.com/openapi-env-test/azure-sdk-for-java (fetch)
10:29:05.892 cmdout 	[generate.py] Checking remote secondary	https://x-access-token:***@github.com/openapi-env-test/azure-sdk-for-java (push)
```

File: eng/common/scripts/TypeSpec-Project-Sync.ps1 (hunk 2 above)
Comment (raych1) [reply eng/common/scripts/TypeSpec-Project-Sync.ps1:106]: I would leave it to user to ensure the path passed in has the right type specs.

**Comment by raych1** [comment]
@weshaggard , I took back my word. The change of adding an optional parameter to sync script isn't a breaking change and shouldn't break current usage. So, it's ok to merge this PR.

**Comment by raych1** [comment]
> @raych1 can you please link the issue you are fixing with this in the description?

Linked the issue in the first comment.

File: eng/common/scripts/TypeSpec-Project-Sync.ps1 (hunk 1 above)
Comment (weshaggard) [reply eng/common/scripts/TypeSpec-Project-Sync.ps1:50]: OK the automation pipeline must be doing something unique when it does its clones as by default origin is always set to what is cloned.

File: eng/common/scripts/TypeSpec-Project-Process.ps1 [hunk 3]
```diff
@@ -71,6 +60,20 @@ function CreateUpdate-TspLocation([System.Object]$tspConfig, [string]$TypeSpecPr
   return $packageDir
 }
 
+function Get-ServiceDir([System.Object]$tspConfig, [string]$repoRoot) {
+  $serviceDir = ""
+  if ($tspConfig["parameters"] -and $tspConfig["parameters"]["service-dir"]) {
+    $serviceDir = $tspConfig["parameters"]["service-dir"]["default"];
+  }
+  else {
+    Write-Error "Missing service-dir in parameters section of tspconfig.yaml."
```
Comment (weshaggard) [inline eng/common/scripts/TypeSpec-Project-Process.ps1:69]: We should probably link to some docs so folks know how to fix this.

**Review by weshaggard — APPROVED** [review]

File: eng/common/scripts/TypeSpec-Project-Process.ps1 (hunk 3 above)
Comment (raych1) [reply eng/common/scripts/TypeSpec-Project-Process.ps1]: I added the sample project tspconfig.yaml for latest schema.

**Pull request merged**